### PR TITLE
MKL-DNN: Add test for conv bias fuse pass

### PR DESCRIPTION
**Pull request by Sand3r-**
Implements C-API Unit Test for correct fusing of convolution and bias when paddle is built with MKL-DNN.

image
Related to: #15615

### CODE REVIEW

File: paddle/fluid/framework/ir/mkldnn/conv_bias_mkldnn_fuse_pass_tester.cc [hunk 1]
```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (sfraczek) [inline paddle/fluid/framework/ir/mkldnn/conv_bias_mkldnn_fuse_pass_tester.cc]: 2019 :)

**Review by sfraczek — CHANGES_REQUESTED** [review]

File: paddle/fluid/framework/ir/mkldnn/conv_bias_mkldnn_fuse_pass_tester.cc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+TEST(ConvBiasFusePass, basic) {
+  auto prog = BuildProgramDesc();
+  std::unique_ptr<ir::Graph> graph(new ir::Graph(prog));
+
+  NaiveExecutor exe{paddle::platform::CPUPlace()};
+  Scope scope;
+  // Init scope, as it is used in pass
+  exe.CreateVariables(prog, 0, true, &scope);
+  graph->Set(kParamScopeAttr, new framework::Scope*(
+                    const_cast<framework::Scope*>(&scope)));
```
Comment (luotao1) [inline paddle/fluid/framework/ir/mkldnn/conv_bias_mkldnn_fuse_pass_tester.cc]: It is not recommended to use `const_cast`.

File: paddle/fluid/framework/ir/mkldnn/conv_bias_mkldnn_fuse_pass_tester.cc (hunk 1 above)
Comment (Sand3r-) [reply paddle/fluid/framework/ir/mkldnn/conv_bias_mkldnn_fuse_pass_tester.cc]: This breaking change has been introduced as of the most recent commit.

File: paddle/fluid/framework/ir/mkldnn/conv_bias_mkldnn_fuse_pass_tester.cc (hunk 2 above)
Comment (Sand3r-) [reply paddle/fluid/framework/ir/mkldnn/conv_bias_mkldnn_fuse_pass_tester.cc]: I've removed the cast since it wasn't even necessary.

**Review by sfraczek — APPROVED** [review]
LGTM

**Pull request merged**